core/src/Cuda/Kokkos_Cuda_Half.hpp

```diff
@@ -290,7 +368,18 @@ class half_t {
 
   template <class T>
   KOKKOS_FUNCTION void operator=(T rhs) volatile {
-    val = cast_to_half(rhs).val;
```

Comment:
Why did this change? Why do you need atomics here now?

Reply:
This is related to my experience with race conditions when using `__half` volatile stores. See https://github.com/kokkos/kokkos/pull/4101#discussion_r681229041.